### DISCUSSION TITLE
How to achieve acrylic background

### DISCUSSION THREAD

**Discussion by minghamn** [fltk-rs/fltk-rs#1324] [Q&A]
I want to implement an acrylic background, but I don't know how I can do it.

**Comment by MoAlyousef** [comment]
If you mean change the window's opacity:
```rust
use fltk::{enums::*, prelude::*, *};

fn main() {
    let app = app::App::default().with_scheme(app::Scheme::Gtk);
    let mut win = window::Window::default().with_size(400, 300);
    win.end();
    win.show();
    win.set_opacity(0.8);
    app.run().unwrap();
}
```
This depends on whether your system supports window transparency or not.

**Reply by minghamn** [reply]
Thank you for your answer.
But this is not the effect I want.
I wish I could have the same effect as this picture.

![image](https://user-images.githubusercontent.com/95327699/205490101-7798bdaa-de37-461d-88a0-aa63e84039d9.png)

**Reply by MoAlyousef** [reply]
From what I've read, windows doesn't expose the api to their acrylic background effect, there are undocumented DWM private apis which you can probably call from Rust since fltk-rs does expose the HWND handle of the window.

A simpler approach is to probably get a png with a blur effect, use it along with set_opacity and you might get some similar effect.

**Reply by minghamn** [reply]
Sorry I accidentally deleted the previous reply.

I found a create that does the acrylic effect.
[window-vibrancy](https://github.com/tauri-apps/window-vibrancy)

But I'm a beginner and I don't know how to use this.
Can you give me a simple example? thank you very much.

**Reply by MoAlyousef** [reply]
You can try:
```toml
# in your Cargo.toml
[dependencies]
fltk = { version = "1.3.22", features=["rwh05"] }
window-vibrancy = "0.3"
```
```rust
use fltk::{prelude::*, *};
use window_vibrancy::apply_acrylic;

fn main() {
    let a = app::App::default().with_scheme(app::Scheme::Gtk);
    app::set_background_color(0x15, 0x22, 0x38);
    let mut w = window::Window::default().with_size(400, 300);
    w.end();
    w.show();

    apply_acrylic(&w, Some((10, 10, 10, 90))).expect("Unsupported platform! 'apply_acrylic' is only supported on Windows");

    a.run().unwrap();
}
```

**Reply by MoAlyousef** [reply]
![image](https://user-images.githubusercontent.com/37966791/205506421-09f193e6-ffc2-4c17-bc35-66e6480551fd.png)

**Reply by minghamn** [reply]
This is the effect I was looking for. 
Thank you !

**Comment by minghamn** [comment]
This doesn't seem to work for me.

My windows version is win11 22H2.
I also turned on the window transparency effect setting.

```
# Cargo.toml
[package]
name = "fltk-acrylic"
version = "0.1.0"
edition = "2021"

See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
fltk = { version = "1.3.22", features = ["rwh05"] }
window-vibrancy = "0.3"

# main.rs
use fltk::{prelude::*, *};
use window_vibrancy::apply_acrylic;

fn main() {
    let a = app::App::default().with_scheme(app::Scheme::Gtk);
    app::set_background_color(0x15, 0x22, 0x38);
    let mut win = window::Window::default().with_size(400, 400);
    win.end();
    win.show();

    apply_acrylic(&win, Some((10, 10, 10, 90))).expect("Unsupported");
    a.run().unwrap();
}
```

Seems to work only for the menu bar.

![image](https://user-images.githubusercontent.com/95327699/205540801-5308a84e-81b1-4d7a-a6bc-1343467d497b.png)

**Reply by MoAlyousef** [reply]
You can try removing the call to app::set_background_color. Also you can try the other effects like apply_blur. window-vibrancy uses undocumented api as far as I'm aware so such api are not guaranteed to work across different versions of windows, or at least not in the same way.

**Reply by minghamn** [reply]
I'll try other methods, thanks for your answer.

**Reply by NoelJacob** [reply]
Hey @hanmingk did you find a workaround?

**Reply by minghamn** [reply]
@NoelJacob I didn't find any other solution. Maybe you could try getting the window handle and then use the windows api to handle the window handle, if you are familiar with the window api.
